Author: Chris Kuehl <ckuehl@ocf.berkeley.edu>
Subject: Add --version option to validator scripts
Forwarded: https://github.com/pre-commit/pre-commit/pull/250
Applied-Upstream: > 0.5.4, https://github.com/pre-commit/pre-commit/commit/ff5e6e5e8
Description: Add `--version' option to validator binaries so that we can
 generate manpages for them using help2man.

--- a/pre_commit/clientlib/validate_base.py
+++ b/pre_commit/clientlib/validate_base.py
@@ -3,6 +3,7 @@
 
 import argparse
 import os.path
+import pkg_resources
 import re
 import sys
 
@@ -71,6 +72,14 @@
         argv = argv if argv is not None else sys.argv[1:]
         parser = argparse.ArgumentParser()
         parser.add_argument('filenames', nargs='*', help=filenames_help)
+        parser.add_argument(
+            '-V', '--version',
+            action='version',
+            version='%(prog)s {0}'.format(
+                pkg_resources.get_distribution('pre-commit').version
+            )
+        )
+
         args = parser.parse_args(argv)
 
         retval = 0
